overlord/snapstate/handlers.go

```diff
@@ -742,6 +747,10 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, tomb *tomb.Tomb) error {
 				Channel:   snapsup.Channel,
 				CohortKey: snapsup.CohortKey,
 				Revision:  snapsup.Revision(),
+				// TODO: do we actually want to use the enforced validation sets
+				// here? i personally think no, since we should have already
+				// validated this change before getting here.
```

Comment:
mmh, that's what you are doing though? what's the real question/TODO here?

Reply:
Was more of a question to highlight during reviews. I"ll remove it, since it should really be unneeded.

Reply:
After some more thought, I think this must remain to stay consistent with our current behavior.

This bit is inside of this branch that is covering some compatibility with an old snapd. I don't think that we'd ever have any validation sets in this case in the first place, but right now we send the enforced validation sets in the case that we do go inside this branch.